<?xml version="1.0"?>
<launch>
  <!-- vehicle info -->
  <arg name="vehicle_info_param_file" default="$(find-pkg-share vehicle_info_util)/config/vehicle_info.param.yaml"/>

  <arg name="input_route_topic_name" default="/planning/mission_planning/route"/>
  <arg name="input_traffic_light_topic_name" default="/perception/traffic_light_recognition/traffic_signals"/>
  <arg name="virtual_traffic_light_topic_name" default="/awapi/tmp/virtual_traffic_light_states"/>
  <arg name="map_topic_name" default="/map/vector_map"/>
  <arg name="behavior_velocity_planner_param_file" default="$(find-pkg-share behavior_velocity_planner)/config/behavior_velocity_planner.param.yaml"/>

  <node pkg="behavior_velocity_planner" exec="behavior_velocity_planner_node" name="behavior_velocity_planner" output="screen">
    <param from="$(var vehicle_info_param_file)"/>
    <remap from="~/input/path_with_lane_id" to="path_with_lane_id"/>
    <remap from="~/input/vector_map" to="$(var map_topic_name)"/>
    <remap from="~/input/vehicle_odometry" to="/localization/kinematic_state"/>
    <remap from="~/input/accel" to="/localization/acceleration"/>
    <remap from="~/input/dynamic_objects" to="/perception/object_recognition/objects"/>
    <remap from="~/input/no_ground_pointcloud" to="/perception/obstacle_segmentation/pointcloud"/>
    <remap from="~/input/traffic_signals" to="$(var input_traffic_light_topic_name)"/>
    <remap from="~/input/virtual_traffic_light_states" to="$(var virtual_traffic_light_topic_name)"/>
    <remap from="~/input/occupancy_grid" to="/sensing/lidar/occupancy_grid"/>
    <remap from="~/output/path" to="path"/>
    <remap from="~/output/stop_reasons" to="/planning/scenario_planning/status/stop_reasons"/>
    <remap from="~/output/infrastructure_commands" to="/planning/scenario_planning/status/infrastructure_commands"/>
    <remap from="~/output/traffic_signal" to="debug/traffic_signal"/>

    <param from="$(var behavior_velocity_planner_param_file)"/>
  </node>
</launch>
